java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -744,6 +744,48 @@ public void sadd_srem_scard_smembers_key_with_non_set_value(BaseClient client) {
         assertTrue(e.getCause() instanceof RequestException);
     }
 
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void smove(BaseClient client) {
+        String setKey1 = "{key}" + UUID.randomUUID();
+        String setKey2 = "{key}" + UUID.randomUUID();
+        String setKey3 = "{key}" + UUID.randomUUID();
+        String nonSetKey = "{key}" + UUID.randomUUID();
+
+        assertEquals(3, client.sadd(setKey1, new String[] {"1", "2", "3"}).get());
+        assertEquals(2, client.sadd(setKey2, new String[] {"2", "3"}).get());
+        // move an elem
+        assertTrue(client.smove(setKey1, setKey2, "1").get());
+        assertEquals(Set.of("2", "3"), client.smembers(setKey1).get());
+        assertEquals(Set.of("1", "2", "3"), client.smembers(setKey2).get());
+        // move an elem which preset at destination
+        assertTrue(client.smove(setKey2, setKey1, "2").get());
+        assertEquals(Set.of("2", "3"), client.smembers(setKey1).get());
+        assertEquals(Set.of("1", "3"), client.smembers(setKey2).get());
+        // move from missing key
+        assertFalse(client.smove(setKey3, setKey1, "4").get());
+        assertEquals(Set.of("2", "3"), client.smembers(setKey1).get());
+        // move to a new set
+        assertTrue(client.smove(setKey1, setKey3, "2").get());
+        assertEquals(Set.of("3"), client.smembers(setKey1).get());
+        assertEquals(Set.of("2"), client.smembers(setKey3).get());
+        // move missing element
+        assertFalse(client.smove(setKey1, setKey3, "42").get());
```

Comment:
how is this different than test on line 767? 
This would be more interesting if setKey3 wasn't already a set.  
```suggestion
        assertFalse(client.smove(setKey1, setKey4, "42").get());
```

Reply:
Added a new test